test/test.ts

```diff
@@ -0,0 +1,67 @@
+import { getVideoUrls } from '../utils';
+import puppeteer from 'puppeteer';
+import assert from 'assert';
+import tmp from 'tmp';
+import fs from 'fs';
+
+let browser: any;
+let page: any;
+
+before(async () => {
+    browser = await puppeteer.launch({
+        headless: true,
+        args: ['--disable-dev-shm-usage']
+    });
+    page = await browser.newPage();
+});
+
+describe('Puppeteer', () => {
+    it('should grab GitHub page title', async () => {
+        await page.goto("https://github.com/", { waitUntil: 'networkidle2' });
+        let pageTitle = await page.title();
+        assert.equal(true, pageTitle.includes('GitHub'));
+
+    }).timeout(15000); // yeah, this may take a while...
+});
+
+after(async () => {
+    await browser.close();
+});
+
+
+describe('Destreamer', () => {
+    it('should parse and sanitize URL list from file', () => {
+        const testIn: string[] = [
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd?",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd&",
+            "",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd?a=b&c",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd?a",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddd",
+            "https://web.microsoftstream.com/video/xxxxxx-zzzz-hhhh-rrrr-dddddddddddd",
+            ""
+        ];
+        const expectedOut: string[] = [
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd?a=b&c",
+            "https://web.microsoftstream.com/video/xxxxxxxx-zzzz-hhhh-rrrr-dddddddddddd?a"
+        ];
+        const tmpFile = tmp.fileSync({ postfix: '.txt' });

```

Comment:
Not familiar with the `tmp` module, what happens if our test fails midway through or gets cancelled (CTRL-C)? Are we leaving the temporary .txt file behind?

Reply:
According to the docs if you don't explicitly tell tmp to keep the file, the temp file will be "closed and unlinked on process exit".
Also this is using the default OS temp dir, no file is being created on our directory :)

Reply:
That was my concern that we're accidentally leaving garbage behind outside our working folder. Let's merge it like this, worry later :) Thanks again!